### ISSUE TITLE
Stance selection on production buildings.

### ISSUE THREAD

**Issue by AoAGeneral** [OpenRA/OpenRA#13817]
Brought up by Anjew in a thread to have production buildings such as barracks and war factories to set their stance of any type. This would allow newly created units to have their new default stance set to this.

**Comment by rob-v** [comment]
With more buildings of the same type, it would be one stance for all these buildings?

**Comment by gdavegdave** [comment]
Yes, this is a good idea. I think it'd be best to be able to assign a production stance to individual buildings. E.g. in TD you may want one barracks to produce infantry on defend and another on attack move. It's trivial to double click to select all barracks / WFs / etc. and the same stance could then be applied to all.

How about having a global option (in settings, perhaps with a configurable hotkey too) to change all production stances. That way, players choose between their preferred stance without having to change it every time they make a new production facility / every game.

**Comment by pchote** [comment]
The most sensible place to manage this is on the production queue.  That means that it is set per-factory in TD, but per-type (infantry/vehicle/etc) in the other mods.  In RA there is no guarantee over which structure a unit will be produced at, and it would be awkward to have units pop out with a different stance just because you placed a new structure immediately before a unit finished construction, or because a stray harvester drives past your main war factory.

**Comment by pchote** [comment]
This isn't a difficult feature, and so while i'm a huge fan of the scope creep this seems like a good compromise to make for the next playtest.  I'll take this on.

**Comment by dan9550** [comment]
A part of me is kind of against this but at the same time i feel like i could live with it. One of those things you need to just try and get a feel for IMO.

Slightly leaning towards one default stance for created units then having the player micro manage new units.

**Comment by airetaM** [comment]
Would be nice if this https://github.com/OpenRA/OpenRA/issues/11727 is being added too.

**Comment by pchote** [comment]
All of our mods have at least one actor that defaults to the HoldFire stance (submarines, spies, stealth tanks, stealth raiders, etc).  This feature can't be implemented without breaking those, and I refuse to touch that change after the personal attacks I received from the community the first time around.

Removing from the milestone and my own future plans.

I pushed a WIP branch of where I got to https://github.com/pchote/OpenRA/tree/production-stances and will eventually file a PR to merge the `*Init` plumbing changes into bleed.

**Comment by rob-v** [comment]
Idea compatible with your WIP:
![image](https://user-images.githubusercontent.com/16348750/29484320-572d9138-84bc-11e7-8329-bef0de5e0b04.png)

When you build different units, you might prefer to keep builtin default stances, but when you are training many Rifle and Rockets and executing some tactic, you could change their default stance optionally and temporarily.

**Comment by rob-v** [comment]
Probably clear - that checkbox in production queue is to enable/disable custom default stances.

**Comment by AndriiYukhymchak** [comment]
Working on it

**Comment by praetp** [comment]
Any update on this ?
